libraries/AP_Scheduler/AP_Scheduler.h

```diff
@@ -93,6 +94,13 @@ class AP_Scheduler
     uint32_t get_loop_period_us() const {
         return 1000000UL / _loop_rate_hz;
     }
+    // get the time-allowed-per-loop:
+    float get_loop_period_s() {
+        if (is_zero(_loop_period_s)) {
+            _loop_period_s = 1.0 / _loop_rate_hz;
+        }
+        return _loop_period_s;
```

Comment:
Why use a member variable? Are you worried that periodic calls will slow things down too much? If we keep it as a variable I think we should:
- initialize the variable on construction (changing the loop rate already means you need to reboot)
- change the microsecond version to also use this variable and just multiply by 1000000 when called.

Reply:
Yes, I'm concerned about affecting performance.  This is called a dozen times at the main loop rate.

Can't initialise at construction time, as we don't need any more race conditions with Parameter object construction (I believe AntennaTracker is still dead on master because of this issue).

I'm not a great fan of doing more operations on floats than necessary.  The microsecond one is nice and inty, and I didn't want to change that.

I'd also have memoised `get_loop_period_micros()` if it wasn't out-of-scope for this change :-)

This isn't a lot of code being encapsulated - and I kind of knew this memoisation would probably raise a flag - bit I think it's worth it.

Reply:
> Can't initialise at construction time, as we don't need any more race conditions with Parameter object construction (I believe AntennaTracker is still dead on master because of this issue).

I wonder where I hit with my head 😄 I wrote initialization first and then changed to construction. Any reason it can't be calculated in `init`?

> I'm not a great fan of doing more operations on floats than necessary. The microsecond one is nice and inty, and I didn't want to change that.

The microsecond one is doing a float division which is much more costly than just a float multiplication (in STM32), hence my suggestion to use the already calculated second version and just multiply on the fly.

> This isn't a lot of code being encapsulated - and I kind of knew this memoisation would probably raise a flag - bit I think it's worth it.

Although I'm not a huge fan of having a variable to store this, I understand your concerns and I'm fine with it - I just think we should make the best use of it 🙂

Reply:
In terms of initialisation... well, there's the fact that the scheduler isn't initialised until after pretty much everything else...

I was going to work around that and memoise both variables.

Instead I've just eliminated the memoisation (patch on top, but should be squashed).

BTW, I can't see how the microsecond one is a float division - everything looks to be an integer to me...

Reply:
@OXINARF ping on this one.

Reply:
> BTW, I can't see how the microsecond one is a float division - everything looks to be an integer to me...

Damn, you are right.

> I was going to work around that and memoise both variables.
>
> Instead I've just eliminated the memoisation (patch on top, but should be squashed).

Humm, aren't you worried about performance anymore? I think it won't be that bad, but I'm wondering if you did any testing or got another opinion.
By the way, I think one thing we can do is use `G_dt` in almost all Copter places where the new method is being used, you need to reboot when loop rate is changed anyway.

Reply:
@OXINARF I decided the performance impact wasn't going to be bad and that simplification in the hopes of getting it merged was more important.

And yeah, there seemed to be oddities where we should've been using `G_dt` IIRC.

So.  Would you like the memoisation back? :-)

Reply:
I don't think I'd like the memoization back, but I'd probably like to see the Copter calls to the method be replaced with `G_dt`, that would certainly alleviate most concerns about this "expensive" call.